### DISCUSSION TITLE
Using lobechat for explaining code with diagrams

### DISCUSSION THREAD

**Discussion by ronald-willems** [lobehub/lobehub#4698] [General Question] [answered]
I'd like to use lobechat to create architecture diagrams based on code repositories. In my job as IT Architect I regularly have to analyze bigger code repos. Creating diagrams with syntaxes like plantUML, mermaid, eraser.io, etc can help quickly understand the current architecture of a code base.

I've already deployed lobechat with database support in Vercel. As a next step I'd like to add a repo to the knowledge base. A repo like lobehub as more than 2000 files. How should I upload this?

I tried to upload as a folder, but it only uploaded around 300 of the files and did not preserve folder structure. 

Then I tried to all text files of a repo into 1 big text file and add that to the knowledge base. However, the 13Mb file is too big to chunk. 

Do you have suggestions?

**Comment by arvinxx** [comment, marked as answer]
If you want to analysis github repo, I think it's better to implement like https://aider.chat . And we do have a plan in the next year to add github repo into knowledge base. Stay tuned!
